src/rendering/SceneManager.cc

```diff
@@ -990,22 +1111,17 @@ rendering::NodePtr SceneManager::NodeById(Entity _id) const
   {
     return vIt->second;
   }
-  else
+  auto lIt = this->dataPtr->lights.find(_id);
+  if (lIt != this->dataPtr->lights.end())
   {
-    auto lIt = this->dataPtr->lights.find(_id);
-    if (lIt != this->dataPtr->lights.end())
-    {
-      return lIt->second;
-    }
-    else
-    {
-      auto sIt = this->dataPtr->sensors.find(_id);
-      if (sIt != this->dataPtr->sensors.end())
-      {
-        return sIt->second;
-      }
-    }
+    return lIt->second;
+  }
+  auto sIt = this->dataPtr->sensors.find(_id);
+  if (sIt != this->dataPtr->sensors.end())
+  {
+    return sIt->second;
   }
+
   return rendering::NodePtr();
```

Comment:
Should we also add code to `SceneManager::NodeById` that searches for a node in [`this->dataPtr->ParticleEmitters`](https://github.com/ignitionrobotics/ign-gazebo/blob/502abfa1a1d2515a71a66bffe86bdae2a06c7e72/src/rendering/SceneManager.cc#L81-L83)? Looking at the code in `ign-rendering`, [a ParticleEmitter virtually inherits a Visual](https://github.com/ignitionrobotics/ign-rendering/blob/8bb934d394fdbd54cbda81b6fb530b68813bfdd9/include/ignition/rendering/ParticleEmitter.hh#L54-L59), and [a VIsual virtually inherits a node](https://github.com/ignitionrobotics/ign-rendering/blob/cac86c83d488b1866374013a92e635ce604a1eee/include/ignition/rendering/Visual.hh#L34-L38). So, I would think that `this->dataPtr->ParticleEmitters` can be searched for nodes as well.

Reply:
@iche033 , can you address this comment?

Reply:
yep done. f58b197aa31669e4323a4e5519dab7fb20dc2df5